pubsublite-beam-io/src/main/java/com/google/cloud/pubsublite/beam/PubsubLiteUnboundedReader.java

```diff
@@ -100,9 +114,32 @@ public PubsubLiteUnboundedReader(
                 permanentError = Optional.of(permanentError.orElse(error));
               }
             });
+    this.backlogReader = backlogReader;
+    this.backlogCache =
+        CacheBuilder.newBuilder()
+            .ticker(ticker)
+            .maximumSize(1)
+            .expireAfterWrite(1, TimeUnit.MINUTES)
+            .refreshAfterWrite(10, TimeUnit.SECONDS)
+            .build(
+                new CacheLoader<Object, Long>() {
+                  public Long load(Object val) throws InterruptedException, ExecutionException {
+                    return computeSplitBacklog().get().getMessageBytes();
+                  }
+                });
```

Comment:
Hmm, I tried a bit but couldn't get it to compile.

It seems like CacheLoader<K,V> is an abstract class, and in order to convert from a lambda it would need to be an interface?

https://guava.dev/releases/22.0/api/docs/com/google/common/cache/CacheLoader.html

Reply:
SG, thanks for the explanation.